analyzers/src/SonarAnalyzer.Common/Common/RuleLoader.cs

```diff
@@ -0,0 +1,37 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2021 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Xml.Linq;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Common
+{
+    public class RuleLoader : IRuleLoader
```

Comment:
I think we can remove the interface and the class itself as it is only used in the AnalyzerConfiguration.cs  it can be a private static method in that class. And for the classes we can use SonarLint.xml files from the Resources instead of mocking.

Reply:
The RuleLoader together with the interface helps to decouple the IO operation and isolate it. 

In this way we can test the `HotspotConfiguration.Initialize` in isolation without accessing the disk and loading an XML file.